exporters/otlp/otlpmetric/internal/otlpmetrictest/otlptest.go

```diff
@@ -55,10 +55,10 @@ func RunEndToEndTest(ctx context.Context, t *testing.T, exp *otlpmetric.Exporter
 	instruments := map[string]data{
 		"test-int64-counter":         {sdkapi.CounterInstrumentKind, number.Int64Kind, 1},
 		"test-float64-counter":       {sdkapi.CounterInstrumentKind, number.Float64Kind, 1},
-		"test-int64-valuerecorder":   {sdkapi.ValueRecorderInstrumentKind, number.Int64Kind, 2},
-		"test-float64-valuerecorder": {sdkapi.ValueRecorderInstrumentKind, number.Float64Kind, 2},
-		"test-int64-valueobserver":   {sdkapi.ValueObserverInstrumentKind, number.Int64Kind, 3},
-		"test-float64-valueobserver": {sdkapi.ValueObserverInstrumentKind, number.Float64Kind, 3},
+		"test-int64-valuerecorder":   {sdkapi.HistogramInstrumentKind, number.Int64Kind, 2},
+		"test-float64-valuerecorder": {sdkapi.HistogramInstrumentKind, number.Float64Kind, 2},
+		"test-int64-valueobserver":   {sdkapi.GaugeObserverInstrumentKind, number.Int64Kind, 3},
+		"test-float64-valueobserver": {sdkapi.GaugeObserverInstrumentKind, number.Float64Kind, 3},
```

Comment:
Yeah, I was thinking of letting this sort of feedback and the possibility of removing `Observer` from the API be left to a future discussion about changing the API from its current state. OTOH, none of the *Observer instruments are keeping their original names here, so possibly it's less disruptive to change `CounterObserver` to `AsyncCounter`, `UpDownCounterObserver` to `AsyncUpDownCounter`, and `GaugeObserver` to `AsyncGauge` **in this PR**?

(I wonder, quietly, if OTel will one day accept the synchronous Gauge, and because of that possibility so I'd leave the name `Gauge` unused.